sql_joins_demystified/sql_joins_demystified.md

```diff
@@ -0,0 +1,589 @@
+
+**Table 2: `subject_address`**
+
+<!-- data-type="none" class="tight-table" style="font-size:80%"-->
+| subj_id  | street_address  | city  | state   | zip  | date_start  | date_end   |
+| :--------- | :--------- | :--------- | :--------- | :--------- | :--------- | :--------- |
+| 11234   | 123 Main Street   | Smithtown    | PA  | 19000    | 2022-01-01   | `NULL`    |
+| 11234   | 123 Oak Lane   | Old Towne    | PA  | 18000   | 2000-01-01    | 2021-12-31    |
+| 93452   | 123 Green Blvd  | Kirby    | TN  | 37000    | 2020-05-01    | `NULL`   |
+
+What you end up with after your successful JOIN will be a single table of results that might just have three columns: `subj_id`, `dep_total`, and `zip`.  That way you can look at the relationship between depression inventory scale and zip code.
+
+You may be asking yourself, "Why not just put all the information together in one table to start with?"  After all, getting one row per subject is how we usually prepare to apply statistical tests. If we could just **start** with one row per subject in SQL, we'd be ready to jump into the science and not have so much data preparation work. 
+
+The reason why this usually doesn't happen in a SQL database, and you instead see lots of different tables that you have to join together is because of a concept known as **normalization**.  To explain normalization, we'll first explain **one-to-many** data relationships.
+
+## One-to-many
+
+Data often has one-to-many relationships. What does one-to-many mean? Consider two use cases, a hospital and a retail website.  Each organization has one-to-many data relationships:
+
+**Hospital one-to-many examples**
+
+* One patient has multiple encounters on different dates
+* One encounter may include multiple orders for different procedures and medications
+* One medication order can give rise to multiple medication administrations
+
+**Retail website one-to-many examples**
+
+* One customer may have several associated addresses for shipping, billing, or both
+* One customer will probably have multiple orders 
+* Each order will have one or many associated items
+
+One-to-many data relationships (one X is related to more than one Y) can be very hard to accommodate in a single table with all the data.  Imagine having a table of order information from an online website, in which a single row contained every item included in that order. 
+
+It would be impossible to know ahead of time how many columns would be needed.  What if we provided enough columns for 100 items in an order to start?  Then, maybe someone orders 110 items.  What do we do?  Also, most orders are much smaller, which means we have many empty cells with no data in them.  It's not an efficient way to store data.  
+
+### Quiz: One-to-Many 
+
+Which of the following relationships are likely to be one-to-many?  Select all the correct answers.
+
+[[X]] Gym member to workout
+[[ ]] Street address to corresponding congressional district
```

Comment:
I initially had a little pause at whether the example of congressional districts might be too USA-centric. But there is another issue with this example because Americans have two sets of congressional districts, one for US Congress and the other for State Legislatures.

Maybe Street address to county? or country? I'm not sure...